### PR TITLE
Add space permissions

### PR DESCRIPTION
**Pull request by motechFR**
Adds the capability to limit actions within a space to specific roles, such as creating a page

### CODE REVIEW

File: components/bounties/BountyList.tsx [hunk 1]
```diff
@@ -48,10 +50,8 @@ export default function BountyList () {
     setSavedBountyFilters(savedBountyFilters.filter(status => BountyStatus[status] !== undefined));
   }, []);
 
-  const isAdmin = useIsAdmin();
-
   // User can only suggest a bounty instead of creating it
-  const suggestBounties = isAdmin === false;
+  const suggestBounties = currentUserPermissions?.createBounty === false;
```
Comment (Devorein) [inline components/bounties/BountyList.tsx:54]: Nice one. We are slowly moving away from admin being super users

File: components/common/CharmEditor/components/inlinePalette/useEditorItems.tsx [hunk 2]
```diff
@@ -24,6 +24,8 @@ import { MAX_EMBED_WIDTH, MIN_EMBED_HEIGHT, MIN_EMBED_WIDTH, VIDEO_ASPECT_RATIO
 import { PluginKey, TextSelection } from 'prosemirror-state';
 import { useMemo } from 'react';
 import DatabaseIcon from '@mui/icons-material/TableChart';
+import { useCurrentSpace } from 'hooks/useCurrentSpace';
```
Comment (Devorein) [inline components/common/CharmEditor/components/inlinePalette/useEditorItems.tsx:27]: This hook is left unused 😛

File: components/settings/roles/spacePermissions/SpacePermissions.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+            <Typography variant='caption'>
+              {targetGroup === 'space' && (
+                'Enabling permissions here will allow every space member to perform the relevant action, whatever their roles and permissions.'
+              )}
+
+              {targetGroup === 'role' && (
+                'Space members with this role can perform all enabled actions, even if these actions are disabled at the space level.'
+              )}
+
+              {targetGroup === 'user' && (
```
Comment (Devorein) [inline components/settings/roles/spacePermissions/SpacePermissions.tsx:160]: Nice we already have partial support for user target groups 🥳

**Review by Devorein — APPROVED** [review]

File: components/settings/roles/spacePermissions/SpacePermissions.tsx (hunk 3 above)
Comment (motechFR) [reply components/settings/roles/spacePermissions/SpacePermissions.tsx:160]: Actually we do have full support for user target groups ;)

When you get round to building user UI, you'll be able to use this SpacePermissions component as a black box without needing to touch any of its implementation

**Pull request merged**